plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/utils/CodeTransformModuleUtils.kt

```diff
@@ -7,14 +7,32 @@ import com.intellij.openapi.module.Module
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.projectRoots.JavaSdkVersion
 import com.intellij.openapi.projectRoots.impl.JavaSdkImpl
+import com.intellij.openapi.roots.LanguageLevelModuleExtensionImpl
 import com.intellij.openapi.roots.ModuleRootManager
 import com.intellij.openapi.roots.ProjectRootManager
 
 /**
- * @description Try to get the module SDK version and fallback to the project SDK version from the "project structure" settings.
+ * @description Try to get the module SDK version and/or Language level from the project settings > modules > source field.
+ * If module inherits from defaults, fallback to the project SDK version from the "project structure" settings.
  */
 fun Module.tryGetJdk(project: Project): JavaSdkVersion? {
-    val sdk = ModuleRootManager.getInstance(this).sdk ?: ProjectRootManager.getInstance(project).projectSdk ?: return null
     val javaSdk = JavaSdkImpl.getInstance()
-    return javaSdk.getVersion(sdk)
+    val moduleRootManager = ModuleRootManager.getInstance(this)
+    val moduleLanguageLevel = this.tryGetJdkLanguageLevelJdk() ?: javaSdk.getVersion(moduleRootManager.sdk!!)
+    val projectSdk = javaSdk.getVersion(ProjectRootManager.getInstance(project).projectSdk!!)
+    return moduleLanguageLevel ?: projectSdk
+}
+
+/**
+ * @description Try to get the project SDK "language level" version from the module "source" settings.
+ * The default value should be set to the project settings SDK and language level, so if the parent SDK is set to
+ * Java 17 and the language level is set to default. The value spit out will be JDK_17. If the parent language
+ * level is set to JDK_1_8 then the default will be JDK_1_8 for the module. You can override all this at the
+ * module "source" setting
+ */
+fun Module.tryGetJdkLanguageLevelJdk(): JavaSdkVersion? {
+    val moduleRootManager = ModuleRootManager.getInstance(this)
+    val languageLevelModuleExtension = moduleRootManager.getModuleExtension(LanguageLevelModuleExtensionImpl::class.java)
+    val languageLevel = languageLevelModuleExtension?.languageLevel
+    return JavaSdkVersion.fromLanguageLevel(languageLevel!!)
```

Comment:
Do we need to update any existing UX messages/documentation that QCT will try to infer sdk version from language level? Also ask customers to ensure their project can build properly when using older language level.

Reply:
Yes, that's another discussion. We don't see this as a one way door as the IDE already trys to set language level.

We mostly need messaging updates around this. In the short term we want to get this out and see if it solves some customer issues that are currently happening.